### DISCUSSION TITLE
Gene expression selection filter configuration

### DISCUSSION THREAD

**Discussion by gauravgadhvi** [vitessce/vitessce-python#230] [Q&A]
Hi Vitessce team,

I am trying to get the gene expression to show on the UMAP for our dataset and the gene selected using the featureList component is not being highlighted on the UMAP. It rather shows expression in all the cells. Can you point to an example or documentation on how to map the gene selection work for showing expression on the UMAP?

Here is my example : 

http://vitessce.io/#?url=data%3A%2C%257B%250A%2520%2520%2522version%2522%253A%2520%25221.0.15%2522%252C%250A%2520%2520%2522name%2522%253A%2520%2522HY%2520Analysis2%2520Demo%2522%252C%250A%2520%2520%2522description%2522%253A%2520%2522Hypothalamus%2520Region%2520Demo%2520single-cell%2520widget%2522%252C%250A%2520%2520%2522datasets%2522%253A%2520%255B%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522uid%2522%253A%2520%2522A%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522name%2522%253A%2520%2522HY2_2023%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522files%2522%253A%2520%255B%250A%2520%2520%2520%2520%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522fileType%2522%253A%2520%2522anndata.zarr%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522url%2522%253A%2520%2522https%253A%252F%252Frobbinsa.me%252Fcelldata%252Fwelchlab%252FHY_allMerged_AnnData_NormData.zarr%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522coordinationValues%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522obsType%2522%253A%2520%2522cell%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522featureType%2522%253A%2520%2522gene%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522featureValueType%2522%253A%2520%2522expression%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522embeddingType%2522%253A%2520%2522UMAP%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522options%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522obsEmbedding%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522path%2522%253A%2520%2522obsm%252FX_umap%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522obsSets%2522%253A%2520%255B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522name%2522%253A%2520%2522CellType%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522path%2522%253A%2520%2522obs%252FhighRes_cellType%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522name%2522%253A%2520%2522Cluster%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522path%2522%253A%2520%2522obs%252FhighRes_cluster%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522name%2522%253A%2520%2522Dataset%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522path%2522%253A%2520%2522obs%252Fdataset%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%255D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522obsFeatureMatrix%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522path%2522%253A%2520%2522X%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522featureFilterPath%2522%253A%2520%2522var%252Fgenes%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%255D%250A%2520%2520%2520%2520%257D%250A%2520%2520%255D%252C%250A%2520%2520%2522initStrategy%2522%253A%2520%2522auto%2522%252C%250A%2520%2520%2522coordinationSpace%2522%253A%2520%257B%250A%2520%2520%2520%2520%2522embeddingType%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522A%2522%253A%2520%2522UMAP%2522%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2522obsType%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522A%2522%253A%2520%2522cell%2522%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2522obsColorEncoding%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522A%2522%253A%2520%2522geneSelection%2522%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2522featureType%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522A%2522%253A%2520%2522gene%2522%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2522featureValueType%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522A%2522%253A%2520%2522expression%2522%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2522featureValueColormapRange%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522A%2522%253A%2520%255B%250A%2520%2520%2520%2520%2520%2520%2520%25200%252C%250A%2520%2520%2520%2520%2520%2520%2520%25200.35%250A%2520%2520%2520%2520%2520%2520%255D%250A%2520%2520%2520%2520%257D%250A%2520%2520%257D%252C%250A%2520%2520%2522layout%2522%253A%2520%255B%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522component%2522%253A%2520%2522scatterplot%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522coordinationScopes%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522obsType%2522%253A%2520%2522A%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522embeddingType%2522%253A%2520%2522A%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522featureType%2522%253A%2520%2522A%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522featureValueType%2522%253A%2520%2522A%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522featureSelection%2522%253A%2520%2522A%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522featureValueColormapRange%2522%253A%2520%2522A%2522%250A%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2522x%2522%253A%25200%252C%250A%2520%2520%2520%2520%2520%2520%2522y%2522%253A%25208%252C%250A%2520%2520%2520%2520%2520%2520%2522w%2522%253A%25206%252C%250A%2520%2520%2520%2520%2520%2520%2522h%2522%253A%25206%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522component%2522%253A%2520%2522obsSets%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522coordinationScopes%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522obsType%2522%253A%2520%2522A%2522%250A%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2522x%2522%253A%25206%252C%250A%2520%2520%2520%2520%2520%2520%2522y%2522%253A%25200%252C%250A%2520%2520%2520%2520%2520%2520%2522w%2522%253A%25206%252C%250A%2520%2520%2520%2520%2520%2520%2522h%2522%253A%25203%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522component%2522%253A%2520%2522obsSetSizes%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522h%2522%253A%25205%252C%250A%2520%2520%2520%2520%2520%2520%2522w%2522%253A%25204%252C%250A%2520%2520%2520%2520%2520%2520%2522x%2522%253A%25208%252C%250A%2520%2520%2520%2520%2520%2520%2522y%2522%253A%25201%252C%250A%2520%2520%2520%2520%2520%2520%2522coordinationScopes%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522obsType%2522%253A%2520%2522A%2522%250A%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522component%2522%253A%2520%2522featureList%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522coordinationScopes%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522featureType%2522%253A%2520%2522A%2522%250A%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2522x%2522%253A%25206%252C%250A%2520%2520%2520%2520%2520%2520%2522y%2522%253A%25203%252C%250A%2520%2520%2520%2520%2520%2520%2522w%2522%253A%25206%252C%250A%2520%2520%2520%2520%2520%2520%2522h%2522%253A%25204%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2522component%2522%253A%2520%2522description%2522%252C%250A%2520%2520%2520%2520%2520%2520%2522coordinationScopes%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2522dataset%2522%253A%2520%2522A%2522%250A%2520%2520%2520%2520%2520%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2522x%2522%253A%25200%252C%250A%2520%2520%2520%2520%2520%2520%2522y%2522%253A%25200%252C%250A%2520%2520%2520%2520%2520%2520%2522w%2522%253A%25206%252C%250A%2520%2520%2520%2520%2520%2520%2522h%2522%253A%25203%250A%2520%2520%2520%2520%257D%250A%2520%2520%255D%250A%257D&edit=false

**Comment by keller-mark** [comment]
Hi @gauravgadhvi 

I think this might be a bug but will need to look into it further.

For now, I think the following configuration does what you are intending, and loads the expression values for a preselected gene

```json
{
  "version": "1.0.15",
  "name": "HY Analysis2 Demo",
  "description": "Hypothalamus Region Demo single-cell widget",
  "datasets": [
    {
      "uid": "A",
      "name": "HY2_2023",
      "files": [
        {
          "fileType": "anndata.zarr",
          "url": "https://robbinsa.me/celldata/welchlab/HY_allMerged_AnnData_NormData.zarr",
          "coordinationValues": {
            "obsType": "cell",
            "featureType": "gene",
            "featureValueType": "expression",
            "embeddingType": "UMAP"
          },
          "options": {
            "obsEmbedding": {
              "path": "obsm/X_umap"
            },
            "obsSets": [
              {
                "name": "CellType",
                "path": "obs/highRes_cellType"
              },
              {
                "name": "Cluster",
                "path": "obs/highRes_cluster"
              },
              {
                "name": "Dataset",
                "path": "obs/dataset"
              }
            ],
            "obsFeatureMatrix": {
              "path": "X"
            }
          }
        }
      ]
    }
  ],
  "initStrategy": "auto",
  "coordinationSpace": {
    "embeddingType": {
      "A": "UMAP"
    },
    "obsType": {
      "A": "cell"
    },
    "obsColorEncoding": {
      "A": "geneSelection"
    },
    "featureType": {
      "A": "gene"
    },
    "featureValueType": {
      "A": "expression"
    },
    "featureSelection": {
      "A": [
        "Gdf5"
      ]
    },
    "featureValueColormapRange": {
      "A": [
        0,
        0.35
      ]
    }
  },
  "layout": [
    {
      "component": "scatterplot",
      "coordinationScopes": {
        "obsType": "A",
        "embeddingType": "A",
        "featureType": "A",
        "featureValueType": "A",
        "featureSelection": "A",
        "featureValueColormapRange": "A",
        "obsColorEncoding": "A"
      },
      "x": 0,
      "y": 8,
      "w": 6,
      "h": 6
    },
    {
      "component": "obsSets",
      "coordinationScopes": {
        "obsType": "A",
        "obsColorEncoding": "A"
      },
      "x": 6,
      "y": 0,
      "w": 6,
      "h": 3
    },
    {
      "component": "obsSetSizes",
      "h": 5,
      "w": 4,
      "x": 8,
      "y": 1,
      "coordinationScopes": {
        "obsType": "A"
      }
    },
    {
      "component": "featureList",
      "coordinationScopes": {
        "obsType": "A",
        "featureType": "A",
        "featureValueType": "A",
        "obsColorEncoding": "A",
        "featureSelection": "A"
      },
      "x": 6,
      "y": 3,
      "w": 6,
      "h": 4
    },
    {
      "component": "description",
      "coordinationScopes": {
        "dataset": "A"
      },
      "x": 0,
      "y": 0,
      "w": 6,
      "h": 3
    }
  ]
}
```

**Reply by gauravgadhvi** [reply]
Hi @keller-mark, thank you for your response. In this case upon initial loading, the UMAP shows the expression of "Gdf5" gene by default. The issue is that when I try to select any other genes from the list of genes in the featureList panel, it doesn't update the UMAP and colors all cells blue. How to make sure that the widget is working upon gene selection and what does the colormap range indicate?
I really appreciate your help! 
Thank you!

Best,
Gaurav
